### PR TITLE
Feature/control plane logging

### PR DESCRIPTION
**Pull request by youngjeong46**
*Issue #, if available:*

*Description of changes:* This PR adds a feature to enable control plane logging with optional parameter.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kcoleman731 — COMMENTED** [review]
LGTM

File: lib/utils/cluster-utils.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+					clusterLogging: [
+						{
+							enabled: true,
+							types: enableLogTypes,
+						},
+					],
+				},
+			},
+		},
+		onDelete: {
```
Comment (askulkarni2) [inline lib/utils/cluster-utils.ts:37]: Is this needed if the cluster is being deleted?

Comment (youngjeong46) [reply lib/utils/cluster-utils.ts:37]: This is in case the user decides to remove control plane logs altogether, but still keep the cluster.

**Review by askulkarni2 — APPROVED** [review]

**Comment by shapirov103** [comment]
I am not sure how this functionality affected the e2e test, but it is consistently failing. It appears to time out on Opa add-on. Have you run provisioning of the blueprint-construct-dev locally?

```


blueprint-construct-dev \|  4/82 \| 12:57:00 AM \| ROLLBACK_COMPLETE    \| AWS::CloudFormation::Stack            \| blueprint-construct-dev
--
 
Failed resources:
blueprint-construct-dev \| 11:34:26 PM \| CREATE_FAILED        \| Custom::AWSCDK-EKS-HelmChart          \| blueprint-construct-dev/chart-gatekeeper/Resource/Default (blueprintconstructdevchartgatekeeper40B0F689) Received response status [FAILED] from custom resource. Message returned: Error: b'Release "ssp-addon-opa-gatekeeper" does not exist. Installing it now.\nError: failed post-install: timed out waiting for the condition\n'
 
Logs: /aws/lambda/blueprint-construct-dev-awscdkawse-Handler886CB40B-W87xml62DofN
 
at invokeUserFunction (/var/task/framework.js:95:19)
at processTicksAndRejections (internal/process/task_queues.js:97:5)
at async onEvent (/var/task/framework.js:19:27)
at async Runtime.handler (/var/task/cfn-response.js:48:13) (RequestId: dc372fb7-524f-4dbe-8de1-e57c7187e81d)
new CustomResource (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/@aws-cdk/core/lib/custom-resource.ts:36:21)
\_ new HelmChart (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/@aws-cdk/aws-eks/lib/helm-chart.ts:69:5)
\_ Cluster.addHelmChart (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/@aws-cdk/aws-eks/lib/cluster.ts:396:12)
\_ Function.KubectlProvider.applyHelmDeployment (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/lib/addons/helm-addon/kubectl-provider.ts:90:36)
\_ KubectlProvider.addHelmChart (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/lib/addons/helm-addon/kubectl-provider.ts:128:32)
\_ OpaGatekeeperAddOn.addHelmChart (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/lib/addons/helm-addon/index.ts:18:32)
\_ OpaGatekeeperAddOn.postDeploy (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/lib/addons/opa-gatekeeper/index.ts:41:28)
\_ /codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/lib/stacks/eks-blueprint-stack.ts:211:22
 
❌  blueprint-construct-dev failed: Error: The stack named blueprint-construct-dev failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE
at Object.waitForStackDeploy (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/lib/api/util/cloudformation.ts:307:11)
at processTicksAndRejections (internal/process/task_queues.js:95:5)
at prepareAndExecuteChangeSet (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/lib/api/deploy-stack.ts:351:26)
at CdkToolkit.deploy (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/lib/cdk-toolkit.ts:194:24)
at initCommandLine (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/bin/cdk.ts:267:9)
The stack named blueprint-construct-dev failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE
Error: The stack named blueprint-construct-dev failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE
at Object.waitForStackDeploy (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/lib/api/util/cloudformation.ts:307:11)
at processTicksAndRejections (internal/process/task_queues.js:95:5)
at prepareAndExecuteChangeSet (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/lib/api/deploy-stack.ts:351:26)
at CdkToolkit.deploy (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/lib/cdk-toolkit.ts:194:24)
at initCommandLine (/codebuild/output/src819050526/src/github.com/aws-quickstart/quickstart-ssp-amazon-eks/node_modules/aws-cdk/bin/cdk.ts:267:9)
make: *** [Makefile:42: deploy-all] Error 1

```

**Comment by youngjeong46** [comment]
> I am not sure how this functionality affected the e2e test, but it is consistently failing. It appears to time out on Opa add-on. Have you run provisioning of the blueprint-construct-dev locally including deleting the stacks with VPC?
> 

I just ran another test and it was successful, including creating, updating and deleting control plane log resource upon provision, then destroying all stacks.

**Comment by youngjeong46** [comment]
@shapirov103 Rebase is complete, PR should be ready to merge pending checks.

**Review by shapirov103 — APPROVED** [review]
lgtm

**Pull request merged**